sycl/unittests/scheduler/SchedulerTestUtils.hpp

```diff
@@ -36,7 +47,14 @@ class MockCommand : public cl::sycl::detail::Command {
     return &MRequirement;
   };
 
-  cl_int enqueueImp() override { return MRetVal; }
+  cl_int enqueueImp() override { return MRetVal; };
+
+  MOCK_METHOD2(enqueue, bool(cl::sycl::detail::EnqueueResultT &,
+                             cl::sycl::detail::BlockingT));
+  bool enqueueOrigin(cl::sycl::detail::EnqueueResultT &EnqueueResult,
+                     cl::sycl::detail::BlockingT Blocking) {
+    return cl::sycl::detail::Command::enqueue(EnqueueResult, Blocking);
+  }
```

Comment:
As far as I understand the mock framework will catch the call to `MockCommand::enqueue` and call `MockCommand::enqueueOrigin` instead. The latter will call `Command::enqueue`. I don't see the point of changes in this part as the result is still the same: `Command::enqueue`  is called anyway.

Reply:
Well, it's the whole point of mocking: to capture function calls. This is the default for all the tests that call `enqueue` method, i.e. call the original implementation while capturing the fact that the method was called. See more here: https://chromium.googlesource.com/external/github.com/google/googletest/+/HEAD/googlemock/docs/cook_book.md#delegating-calls-to-a-parent-class

Reply:
I understand the point of mocking in general. Though I wonder if you need to mock `Command::enqueue` method at all? Did I miss something in the patch?

Reply:
This is how I test the changes. See this change for example https://github.com/intel/llvm/pull/2347/commits/794570d51714c7865044d6e55918bbcb64f2ed16. I think it is a good way to write descriptive and small tests.

Reply:
No objections, then.